sklearn/pipeline.py

```diff
@@ -10,22 +10,38 @@
 # License: BSD
 
 from collections import defaultdict
+import time
 
 import numpy as np
 from scipy import sparse
 
 from .base import clone, TransformerMixin
 from .externals.joblib import Parallel, delayed
 from .externals import six
-from .utils.metaestimators import if_delegate_has_method
+from .utils.metaestimators import if_delegate_has_method, _BaseComposition
```

Comment:
revert to the version of master with line 20 and line 24. It is equivalent to master in bring 2 diffs.